scheduler/src/cook/mesos/constraints.clj

```diff
@@ -211,12 +218,18 @@
    sequence of Fenzo TaskTracker.ActiveTasks, which correspond to ALL the running tasks in group.
    This includes both running tasks and tasks that are being assigned by Fenzo
    in the current cycle."
-  [db group task-id same-cycle-task-ids task-tracker-state]
-  (let [cotask-ids (disj
-                     (clojure.set/union
-                       (group/group->running-task-id-set db group) ; Tasks that are in the database (running or scheduled)
-                       same-cycle-task-ids) ; Tasks in same cycle
-                     task-id)] ; Do not include this task-id
+  [db group task-id same-cycle-task-ids task-tracker-state running-task-ids-cache]
```

Comment:
Can we not reuse the same cache (i.e. the `cotask-cache` from `get-cotasks`) and use a `map` to extract the task ids?